lib/Model/Type.cpp

```diff
@@ -821,12 +885,18 @@ RecursiveCoroutine<bool> QualifiedType::verify(VerifyHelper &VH) const {
     if (HasNext and Q.isConstQualifier() and NextQIt->isConstQualifier())
       rc_return VH.fail();
 
+    // Check that the Array Qualifiers are a contiguous trailing block.
+    if (FoundArray and not Q.isArrayQualifier())
+      rc_return VH.fail();
```

Comment:
Didn't we decide for artificial typedefs? Or not for this thing?

Reply:
Not for this thing. Artificla typedefs were for returning (or passing) arrays by copy from (into) functions.

Reply:
Uhm, ACK, not sure what's the criterion here though.

Reply:
The criterion here is explained in the commit message of the commit that introduces the change:
> With this commit, `model::QualifiedType`s that have at least one `Array` `Qualifier`, require all the `Array` `Qualifier`s to be contiguous and packed at the end of the vector of qualifiers. This mimics what C does.

I thought we had agreed on this.

Reply:
Ok, for now, don't include this commit in the merge. I'll take it out of this merge request for now. I'll see what I can do on `revng-c` side to prevent the need for this.

Reply:
Took this commit out for now. I'll re-submit it later if it turns out to be really needed.